### PR TITLE
Feature/tracks

### PR DESCRIPTION
**Pull request by mindovermiles262**

### CODE REVIEW

File: app/models/track.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class Track < ApplicationRecord
   has_many :users
-  has_many :track_courses
+  has_many :track_courses, dependent: :destroy
```
Comment (arku) [inline app/models/track.rb]: What do you think about using `dependent: :delete_all` instead?

I don't think we have any callbacks on `TrackCourse`. So there's no need to instantiate objects and delete them one by one. We can delete them all with a single `DELETE` SQL query.

File: app/controllers/tracks_controller.rb [hunk 2]
```diff
@@ -11,6 +11,6 @@ def index
   private
 
   def ordered_decorated_courses(track_courses)
```
Comment (arku) [inline app/controllers/tracks_controller.rb]: Why do we need to pass the `track_courses` as an argument?

Is there any reason not to use `@track.courses` inside `ordered_decorated_courses` method?

File: app/models/track.rb (hunk 1 above)
Comment (arku) [inline app/models/track.rb]: What do you think about using `dependent: :delete_all` instead?

I don't think we have any callbacks on `TrackCourse`. So there's no need to instantiate objects and delete them one by one. We can delete them all with a single `DELETE` SQL query.

File: db/seeds/08_full_stack_track.rb [hunk 3]
```diff
@@ -7,8 +7,18 @@
   Track.delete_all
 end
 
-Track.create!(
+create_or_update_track(
   title: "Full Stack",
   description: "The OG Odin",
-  position: 1
+  position: 1,
+  courses: [
```
Comment (arku) [inline db/seeds/08_full_stack_track.rb]: @mindovermiles262 The track courses created don't have their `track_position`s set. Can you add those?

File: app/models/track.rb (hunk 1 above)
Comment (mindovermiles262) [reply app/models/track.rb]: That will work much better. We do not need to instantiate each object before deleting. Changed to use `delete_all`

**Pull request merged**